fuel/datasets/cifar10.py

```diff
@@ -40,46 +33,31 @@ class CIFAR10(IndexableDataset):
 
     """
     provides_sources = ('features', 'targets')
-    folder = 'cifar10'
-    files = {
-        'train': [os.path.join('cifar-10-batches-py',
-                               'data_batch_{}'.format(i))
-                  for i in range(1, 6)],
-        'test': ['cifar-10-batches-py/test_batch']
-    }
+    filename = 'cifar10.hdf5'
 
-    def __init__(self, which_set, flatten=True, **kwargs):
+    def __init__(self, which_set, flatten=True, dtype=config.floatX,
+                 **kwargs):
         if which_set not in ('train', 'test'):
             raise ValueError("CIFAR10 only has a train and test set")
 
+        kwargs.setdefault('load_in_memory', True)
         self.which_set = which_set
-        self.flatten = flatten
+        self.reshape = flatten
+        self.dtype = dtype
```

Comment:
No need to make this an attribute. For the moment it's sufficient that you systematically use `fuel.config.floatX`, so you can just use that directly further down.